File: edexOsgi/build.edex/opt/esb/js/ProfilerRequest.js
/**
 * Performs a TermQuery for Profiler data.
 * 
 * Usage:
 *   TBD
 *   
 * SOFTWARE HISTORY
 *
 * Date         Ticket#    Engineer    Description
 * ------------ ---------- ----------- --------------------------
 * 20080310            969 jkorman     Initial implementation.
 */
function ProfilerRequest(){
  this.plugin = "profiler";
  this.subscribe = false;
  this.subscription = null;
  this.queryResults = null;
  this.asciiResponse = false;
  this.query = new TermQuery(this.plugin, subscriptionDataFieldId, subscriptionDataQueryId);
}

function _addParameter(name,value,operand){
    if(arguments.length==2){
        this.query.addParameter(name,value);
    } else{
        this.query.addParameter(name,value,operand);
    }
}

function _addList(name, value){
  this.query.addList(name, value);
}

function _setCount(count){
  this.query.setCount(count);
}

function _setSortValue(sortValue){
  this.query.setSortBy(sortValue);
}

function _enableSubscription(){
  this.subscribe = true;
}

function _enableAsciiResponse(){
  this.asciiResponse = true;
}

function _execute()
{
  if(this.subscribe){
    this.subscription = new Subscription();
    this.subscription.setup(this.query);
  }
  
  this.queryResults = this.query.execute();
  if(this.queryResults == null || this.queryResults.size() == 0)
  {
    var response = new MakeResponseNull("Query returned 0 results.",
                                            this.query);
    return response.execute();
  }
  else
  {
    if(this.asciiResponse)
    {
       return this.makeAsciiResponse();
    }
    else
    {
       return this.makeXmlResponse();
    }
  }
}

function _makeXmlResponse()
{
  var xmlResults = new Array();
  var response = new Array();
  for(i=0; i < this.queryResults.size(); i++)
  {    
    var makeResponse = new MakeResponseXml(this.queryResults.get(i));
    response[i] = makeResponse.execute();
  }
  return response;
}

function _makeAsciiResponse()
{
  var xmlResults = new Array();
  var response = new Array();
  for(i=0; i < this.queryResults.size(); i++)
  {    
    var toXml = new DataToXml(this.queryResults.get(i)); 
    xmlResults[i] = toXml.execute();     
    var makeResponse = new MakeResponseAscii(this.queryResults.get(i), xmlResults[i]);
    response[i] = makeResponse.execute(); 
  }
  return response;
}

ProfilerRequest.prototype.execute = _execute;
ProfilerRequest.prototype.makeXmlResponse = _makeXmlResponse;
ProfilerRequest.prototype.makeAsciiResponse = _makeAsciiResponse;
ProfilerRequest.prototype.addParameter = _addParameter;
ProfilerRequest.prototype.addList = _addList;
ProfilerRequest.prototype.setCount = _setCount;
ProfilerRequest.prototype.setSortValue = _setSortValue;
ProfilerRequest.prototype.enableSubscription = _enableSubscription;
ProfilerRequest.prototype.enableAsciiResponse = _enableAsciiResponse;

File: edexOsgi/com.raytheon.edex.uengine/utility/edex_static/base/python/ProfilerRequest.py
##
##

##
# This is a base file that is not intended to be overridden.
##

##
# uengine is deprecated and will be removed from the system soon. Migrate your
# apps to using the Data Access Framework (DAF).
##

#
# Request of metar records (provides interface to pointdata)
#
#    
#     SOFTWARE HISTORY
#    
#    Date            Ticket#       Engineer       Description
#    ------------    ----------    -----------    --------------------------
#    08/19/09                      jkorman        Initial Creation - Recoded from ObsRequest.py
#    
# 



import PointDataQuery
from java.util import ArrayList
from com.raytheon.uf.common.dataplugin.profiler.dao import ProfilerDataTransform
from com.raytheon.uf.common.message.response import ResponseMessageGeneric


class ProfilerRequest(PointDataQuery.PointDataQuery):
    
    def __init__(self, pluginNameDummyVar):
        PointDataQuery.PointDataQuery.__init__(self, "profiler")

    def addParameter(self, name, value, operand="=",className=None):    
        self._pdq.addParameter(name, value, operand)
    
    def addList(self, name, value,className=None):
        self._pdq.addParameter(name, value, "in")

    def setCount(self, count):
        # self._pdq.setCount(count)
        pass
 
    def setSortValue(self, sortValue,order,className=None):
        self._pdq.setSortBy(sortValue,order,className)
        
    def setOrderByList(self,orderList,ascending,className=None):
        self._pdq.setSortBy(orderList, ascending,className)


    def execute(self):
        self.requestAllLevels()
        self._pdq.setParameters(ProfilerDataTransform.MAN_PARAMS_LIST)        
        self.queryResults = self._pdq.execute()
        if self.queryResults is None:
            return self.makeNullResponse()
        else:
            records = ProfilerDataTransform.toProfilerRecords(self.queryResults)
            size = len(records)
            response = ArrayList()
            for i in range(size):
                response.add(ResponseMessageGeneric(records[i]))
            return response
    